beancount/parser/booking_full.py

```diff
@@ -145,6 +145,10 @@ def _book(entries, options_map, methods):
 
             # Resolve reductions to a particular lot in their inventory balance.
             repl_postings = []
+            # This local balance tracks changes while processing this transaction.
```

Comment:
Is this new? I suspect this is going to make booking much slower.
See comment below.

Reply:
So I timed it and I'm surprised it doesn't.
Let's merge it.
I think the next version we'll have to review and define inputs and outputs better so we don't have to compute partial balances like this.

Reply:
```
bergamot [git|v2]:~/p/beancount$ hyperfine -i -m 25  -- "bean-check -C $L"
Benchmark #1: bean-check -C /home/blais/r/q/office/accounting/blais.beancount
  Time (mean ± σ):      5.617 s ±  0.971 s    [User: 5.485 s, System: 0.085 s]
  Range (min … max):    4.609 s …  7.947 s    25 runs

  Warning: Ignoring non-zero exit code.
bergamot [git|master]:~/p/beancount$ cat /tmp/after 
bergamot [git|master]:~/p/beancount$ hyperfine -i -m 25  -- "bean-check -C $L"
Benchmark #1: bean-check -C /home/blais/r/q/office/accounting/blais.beancount
  Time (mean ± σ):      5.647 s ±  0.455 s    [User: 5.501 s, System: 0.090 s]
  Range (min … max):    4.709 s …  6.406 s    25 runs
```

Reply:
It's new the the revision I did when making the average cost calculator avoid mutating the inventories.
Thanks for checking the latency. I agree on doing some work here in the future. I was thinking about how to do lot transfer for the crypto case and making booking more configurable in general. An example that would be nice is being able to write a tax optimizing booker that auto selects losses before gains. Both of those need clarity